src/pyscaffold/extensions/interactive.py

```diff
@@ -0,0 +1,282 @@
+    There are no public methods that allow inspection/reflection, and therefore in order
+    to do so, one has to rely in a few non-public methods (according to Python's
+    convention, the ones starting with a ``_`` symbol).
+    Since :mod:`argparse` implementation is very stable and mature, these non-public
+    method are very unlikely to change and, therefore, it is relatively safe to use
+    these methods, however developers and maintainers have to be aware and pay attention
+    to eventual breaking changes.
+    The non-public functions are encapsulated in the functions :obj:`get_actions` and
+    :obj:`format_args` in this file, in order to centralise the usage of non-public API.
+"""
+import os
+import shlex
+import textwrap
+from argparse import Action, ArgumentParser
+from collections import abc
+from functools import lru_cache, reduce
+from itertools import chain
+from typing import List, Optional, Set
+
+from .. import api, cli, file_system, shell, templates
+from ..actions import ScaffoldOpts as Opts
+from ..actions import get_default_options
+from . import Extension
+from . import list_from_entry_points as list_all_extensions
+
+INDENT_LEVEL = 4
+HEADER = templates.get_template("header_interactive")
+
+
+CONFIG_KEY = "interactive"
+CONFIG = {
+    "ignore": ["--help", "--version"],
+    "comment": ["--verbose", "--very-verbose"],
+}
+"""Configuration for the options that are not associated with an extension class.
+This dict value consist of a set of metadata organised as follows:
+
+- Each value must be a list of "long" :obj:`argparse` option strings (e.g. `"--help"`
+  instead of `"-h"`).
+- Each key implies on a different interpretation for the metadata:
+
+    - ``"ignore"``: Options that should be simply ignored when creating examples
+    - ``"comment"``: Options that should be commented when creating examples,
+      even if they appear in the original :obj:`sys.argv`.
+
+Extension classes (or instances) can also provide configurations by defining a
+``interactive`` attribute assigned to a similar :obj:`dict` object.
+"""
+
+
+@lru_cache(maxsize=2)
+def get_config(kind: str) -> Set[str]:
+    """Get configurations that will be used for generating examples
+    (from both :obj:`CONFIG` and the ``interactive`` attribute of each extension).
+
+    The ``kind`` argument can assume the same values as the :obj:`CONFIG` keys.
+    """
+    # TODO: when `python_requires >= 3.8` use Literal["ignore", "comment"] instead of
+    #       str for type annotation of kind
+    configurable = CONFIG.keys()
+    assert kind in configurable
+    initial_value = set(CONFIG[kind])
+    fallback_config: dict = {k: [] for k in configurable}
+
+    def _reducer(acc, ext):
+        config_from_ext = getattr(ext, CONFIG_KEY, fallback_config)
+        return acc | set(config_from_ext.get(kind, []))
+
+    return reduce(_reducer, list_all_extensions(), initial_value)
```

Comment:
General note: It always takes me several minutes to understand what's happening in a `reduce` function and I think I am not the only one as Guido von Rossum once stated something similar. At that point, I have no better idea and this functional style is really smart but makes it also hard to understand what's going on.

Reply:
I can revert it to a more procedural approach.

The main idea is to collect all the `interactive` configuration dicts from the extensions and merge them with `CONFIG`.

Reply:
Or you add your last sentence as a comment just before the return statement to make it easier for future contributors to understand what's going on.